Combobox examples with listbox popup: Add properties and states to listbox control button

jongund opened the pull request:
Fixes #1333 and #1331:
* Updates button to include aria attributes identified in issue #1333 and issue #1331  
* Adds regression test for the newly added attributes.

#### Preview link

[View the combobox examples in the compare branch with RawGitHack](https://raw.githack.com/w3c/aria-practices/issue1333-add-aria-to-button/examples/combobox/combobox-autocomplete-both.html)

### Review checklist

* [x] [Editorial review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#editorial_review) by Matt (@mcking65)
* [x] [Functional review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#functional_review) by Matt (@mcking65)
* [x] [Visual design review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#design_review) by Simon (@zcorpan)
* [x] [Accessibility review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#accessibility_review) by Matt (@mcking65)
* [x] [Code review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#code_review) by Simon (@zcorpan)
* [x] [Test review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#test_review) by Simon (@zcorpan)

mcking65 commented on examples/combobox/combobox-autocomplete-both.html:
```diff
@@ -51,16 +51,16 @@ <h2 id="ex_label">Example</h2>
-              aria-expanded="false" aria-controls="lb1">
-            <button type="button" id="cb1-button" aria-label="Open" tabindex="-1">
+              aria-expanded="false" aria-controls="cb1-listbox">
+            <button type="button" id="cb1-button" aria-label="State" aria-expanded="false" aria-controls="cb1-listbox" tabindex="-1">
```
The name of the listbox is "States" plural, and this button controls that, so I recommend:

```suggestion
            <button type="button" id="cb1-button" aria-label="States" aria-expanded="false" aria-controls="cb1-listbox" tabindex="-1">
```

jongund replied:
Fixed

mcking65 commented on examples/combobox/combobox-autocomplete-list.html:
```diff
@@ -53,7 +53,7 @@ <h2 id="ex_label">Example</h2>
             <input id="cb1-input" class="cb_edit" type="text" role="combobox" aria-autocomplete="list"
               aria-expanded="false" aria-controls="cb1-listbox">
-            <button id="cb1-button" tabindex="-1" aria-label="Open">
+            <button id="cb1-button" tabindex="-1"  aria-label="State" aria-expanded="false" aria-controls="cb1-listbox">
```
Same recommendation as above:

```suggestion
            <button id="cb1-button" tabindex="-1"  aria-label="States" aria-expanded="false" aria-controls="cb1-listbox">
```

jongund replied:
Fixed

mcking65 commented on examples/combobox/combobox-autocomplete-none.html:
```diff
@@ -53,7 +53,7 @@ <h2 id="ex_label">Example</h2>
                    aria-expanded="false"
                    aria-controls="cb1-listbox">
-            <button type="button" id="cb1-button" tabindex="-1" aria-label="Open">
+            <button type="button" id="cb1-button" tabindex="-1"  aria-label="State" aria-expanded="false" aria-controls="cb1-listbox" >
```
This one controls the "Previous Searches" list, so the aria-label should reflect that.

```suggestion
            <button type="button" id="cb1-button" tabindex="-1"  aria-label="Previous Searches" aria-expanded="false" aria-controls="cb1-listbox" >
```

jongund replied:
Fixed

mcking65 commented on examples/combobox/combobox-autocomplete-both.html:
```diff
@@ -426,6 +426,44 @@ <h3 id="rps_label_listbox">Listbox Popup</h3>
+          </tr>
+        </thead>
+        <tbody>
+          <tr data-test-id="button-aria-controls">
```
I think we need a row before this one for aria-label

jongund replied:
Fixed

mcking65 commented on examples/combobox/combobox-autocomplete-list.html:
```diff
@@ -422,8 +422,44 @@ <h3 id="rps_label_listbox">Listbox Popup</h3>
+          </tr>
+        </thead>
+        <tbody>
+          <tr data-test-id="button-aria-controls">
```
I think we need a row before this one for aria-label

jongund replied:
Fixed

mcking65 commented on examples/combobox/combobox-autocomplete-none.html:
```diff
@@ -371,8 +371,43 @@ <h3 id="rps_label_listbox">Listbox Popup</h3>
+          </tr>
+        </thead>
+        <tbody>
+          <tr data-test-id="button-aria-controls">
```
I think we need a row before this one for aria-label

jongund replied:
Fixed

jongund commented:
@mcking65 

I updated the pull request with your suggestions and also added a row for `tabindex=-``.  I also added a section in the keyboard support section saying that the `button` is being removed from tab sequence of the page, but is still important for assistive technologies that use touch events to open the listbox.

jongund commented:
@mcking65 

I updated this pull request to also address issue #1331 high contrast mode.

zcorpan commented:
Visual design review: the relevant change is about Windows high contrast, reviewed in https://github.com/w3c/aria-practices/issues/1331#issuecomment-594011492

zcorpan commented on examples/combobox/combobox-autocomplete-list.html:
```diff
@@ -53,10 +53,11 @@ <h2 id="ex_label">Example</h2>
             <input id="cb1-input" class="cb_edit" type="text" role="combobox" aria-autocomplete="list"
               aria-expanded="false" aria-controls="cb1-listbox">
-            <button id="cb1-button" tabindex="-1" aria-label="Open">
+            <button id="cb1-button" tabindex="-1"  aria-label="States" aria-expanded="false" aria-controls="cb1-listbox">
```
```suggestion
            <button id="cb1-button" tabindex="-1" aria-label="States" aria-expanded="false" aria-controls="cb1-listbox">
```

zcorpan commented on test/tests/combobox_autocomplete-both.js:
```diff
@@ -149,6 +149,54 @@ ariaTest('"aria-selected" attribute on options element', exampleFile, 'option-ar
+
+  t.false(
+    await popupElement.isDisplayed(),
+    'Popup element should not be displayed when \'aria-expanded\' is false\''
```
```suggestion
    'Popup element should not be displayed when \'aria-expanded\' is \'false\''
```

zcorpan commented on test/tests/combobox_autocomplete-both.js:
```diff
@@ -149,6 +149,54 @@ ariaTest('"aria-selected" attribute on options element', exampleFile, 'option-ar
+
+  t.true(
+    await popupElement.isDisplayed(),
+    'Popup element should be displayed when \'aria-expanded\' is true\''
```
```suggestion
    'Popup element should be displayed when \'aria-expanded\' is \'true\''
```

zcorpan commented on test/tests/combobox_autocomplete-none.js:
```diff
@@ -154,6 +155,75 @@ ariaTest('"aria-selected" attribute on options element', exampleFile, 'option-ar
+
+  t.false(
+    await popupElement.isDisplayed(),
+    'Popup element should not be displayed when \'aria-expanded\' is false\''
```
```suggestion
    'Popup element should not be displayed when \'aria-expanded\' is \'false\''
```

zcorpan commented on test/tests/combobox_autocomplete-none.js:
```diff
@@ -154,6 +155,75 @@ ariaTest('"aria-selected" attribute on options element', exampleFile, 'option-ar
+
+  t.true(
+    await popupElement.isDisplayed(),
+    'Popup element should be displayed when \'aria-expanded\' is true\''
```
```suggestion
    'Popup element should be displayed when \'aria-expanded\' is \'true\''
```

zcorpan commented on examples/combobox/combobox-autocomplete-none.html:
```diff
@@ -53,7 +53,7 @@ <h2 id="ex_label">Example</h2>
                    aria-expanded="false"
                    aria-controls="cb1-listbox">
-            <button type="button" id="cb1-button" tabindex="-1"  aria-label="State" aria-expanded="false" aria-controls="cb1-listbox" >
+            <button type="button" id="cb1-button" tabindex="-1"  aria-label="Previous Searches" aria-expanded="false" aria-controls="cb1-listbox" >
```
```suggestion
            <button type="button" id="cb1-button" tabindex="-1" aria-label="Previous Searches" aria-expanded="false" aria-controls="cb1-listbox" >
```

zcorpan commented:
Maybe https://github.com/w3c/aria-practices/pull/1343 should be merged first, and this be rebased on top and remove the `t.plan(1)`s.

zcorpan commented:
> Maybe #1343 should be merged first, and this be rebased on top and remove the `t.plan(1)`s.

Never mind; I think that PR should add a lint check, which means other PRs like this one don't need to be blocked on it.

zcorpan commented:
`npm run regression-report` gives

```
combobox/combobox-autocomplete-both.html:
    combobox-id
combobox/combobox-autocomplete-list.html:
    combobox-id
combobox/combobox-autocomplete-none.html:
    combobox-id
```

I think those failures were introduced by #1276. I can report a new issue about this.

zcorpan commented:
Actually I can't merge, I'm not authorized to push to protected branches.
